File: android/src/ti/airship/events/PushReceivedEvent.java
/* Copyright Airship and Contributors */

package ti.airship.events;

import androidx.annotation.NonNull;

import java.util.HashMap;
import java.util.Map;

import ti.airship.AirshipTitaniumModule;
import ti.airship.TiPush;

public class PushReceivedEvent implements Event {
    private final TiPush tiPush;
    private final boolean receivedInForeground;

    public PushReceivedEvent(@NonNull TiPush tiPush, boolean receivedInForeground) {
        this.tiPush = tiPush;
        this.receivedInForeground = receivedInForeground;
    }

    @NonNull
    @Override
    public Map<String, Object> getData() {
        Map<String, Object> map = new HashMap<>(tiPush.toMap());
        map.put("receivedInForeground", receivedInForeground);
        return map;
    }

    @NonNull
    @Override
    public String getName() {
        return AirshipTitaniumModule.EVENT_PUSH_RECEIVED;
    }
}

File: android/src/ti/airship/events/PushReceivedEvent.kt
/* Copyright Airship and Contributors */

package ti.airship.events

import com.urbanairship.push.NotificationInfo
import com.urbanairship.push.PushMessage
import ti.airship.TiAirshipModule

class PushReceivedEvent : Event {
    override val data: Map<String, Any>
    override val name: String = TiAirshipModule.eventPushReceived

    constructor(pushMessage: PushMessage) {
        data = pushMessage.eventPayload()
    }

    constructor(notificationInfo: NotificationInfo) {
        data = notificationInfo.eventPayload()
    }
}